ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/shared/service/internal/PermissionsServiceImpl.java

```diff
@@ -45,6 +66,23 @@ public PermissionsServiceImpl(AuthenticationService authenticationService,
 
   @Override
   public String getFilePermissionsEntity(String stableId) {
+    
+    if (authenticationService.getAuthentication() instanceof OAuth2Authentication) {
+        OAuth2Authentication authentication = (OAuth2Authentication) authenticationService.getAuthentication();
```

Comment:
As the dataset-file mapping is stored in this solution in the authentication object. Would it make sense to add a function in the authenticationService to retrieve them instead of doing all these checks here?

Reply:
Done. Moved the code inside AuthenticationService.